wurst/systems/trade/GoldTrading.wurst

```diff
@@ -0,0 +1,50 @@
+package GoldTrading
+import LinkedList
+import ChatCommands
+import SimError
+import PlayerExtensions
+import StringExtensions
+import PlayerUtils
+
+CommandHandler giveGold = (player giver, LinkedList<string> args) -> begin
+    let tribeId = giver.getTribeId()
+    let playerId = findPlayerIdByStringMatchingTribe(args.get(1), tribeId)
+    let amount = args.get(2).toInt()
+    let currentGold = giver.getLumber()
+    let target = players[playerId]
+
+    if playerId < 0 or playerId > 11
+        simError(giver, "Command: -g player amount")
+    else if amount <= 0
+        simError(giver, "Amount must be greater than zero")
+    else if amount > currentGold
+        simError(giver, "Not enough gold")
+    else if playerId == giver.getId()
+        simError(giver, "You cannot trade with yourself")
+    else if target.getTribeId() != tribeId
+        simError(giver, "You cannot trade with that player")
+    else
+        giver.subLumber(amount)
+        target.addLumber(amount)
+
+        let targetMessage = (
+            "You received ".color(GENERAL_COLOR) +

```

Comment:
I'm guessing it can't be supported due to limits on variable amount of arguments, but does Wurst not have anything sort of string formatting in their standard library? If not, this should still be aligned better.

Reply:
Line is too long if I put it all together. How would you format it? Also, Wurst unfortunately doesn't have string interpolation (though I think it would be easy to add in a fashion similar to what JavaScript did).

Reply:
Align the outermost plus signs with the first equal. I'd personally also break up the nest concatenation so that each line is a single piece with a single color.

Reply:
Can you post a snippet? Like

```
let giverMessage = (
  "Sent ".color(GENERAL_COLOR) +
  args.get(2).color(GOLD_COLOR) +
  " gold".color(GOLD_COLOR) +
  " to ".color(GENERAL_COLOR) +
  target.getNameColored()
)
```

Reply:
https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code-and-syntax-highlighting

Reply:
So we can do 

```
let giverMessage = "{0}Sent|r {1}{2} gold|r{3} to |r{4}".format(GENERAL_COLOR, GOLD_COLOR, args.get(2), GENERAL_COLOR, target.getNameColored())
```

Reply:
I was suggesting something like:
    let giverMessage = "Sent ".color(GENERAL_COLOR)
                                + args.get(2).color(GOLD_COLOR)
                                + " gold".color(GOLD_COLOR)
                                + " to ".color(GENERAL_COLOR)
                                + target.getNameColored()
Definitely prefer `format` though.